libs/blockchain-watcher/feature-app/src/lib/service/ethService.ts

```diff
@@ -1,63 +1,103 @@
 import { ethers } from 'ethers'
 import { mailService } from './mailService'
 import { envStore } from '../store/envStore'
+import {
+  SlmChargeback__factory,
+  SlmEscrow__factory,
+  SlmFactory,
+  SlmFactory__factory,
+  SlmPreorder__factory,
+} from '@solomon/shared/util-contract'
+import { Provider } from '@ethersproject/providers'
+import { deliverService } from './deliverService'
 
 export class EthService {
-  provider = null as any
-  contract = null as any
+  provider: Provider | null = null
+  contract: SlmFactory | null = null
 
-  async onChargebackCreated() {
-    // TODO: Process event
-    await mailService.sendChargebackCreatedEmail('A')
-    await mailService.sendChargebackCreatedEmail('B')
+  async onChargebackCreated(chargebackAddress: string) {
+    if (!this.provider) {
+      return
```

Comment:
If we don't have a provider, should we throw an error rather than silently fail?

Reply:
You are right. Throw error would be better than silence fail.